pinecone/client_test.go

```diff
@@ -27,6 +29,47 @@ func TestClient(t *testing.T) {
 	suite.Run(t, new(ClientTests))
 }
 
+func TestHandleErrorResponseBody(t *testing.T) {
```

Comment:
These tests are attempting to validate the parsing of various `http.Response` bodies:
- `ErrorResponse`
- generic JSON
- `string`